src/audio/copier/copier.c

```diff
@@ -944,7 +944,7 @@ static int copier_reset(struct comp_dev *dev)
 	for (i = 0; i < cd->endpoint_num; i++) {
 		if (dev->ipc_config.type == SOF_COMP_HOST && !cd->ipc_gtw) {
 			if (cd->hd->chan)
-				notifier_unregister(cd->endpoint[i],
+				notifier_unregister(dev,
```

Comment:
does the notifier register need to be inside the loop? I think not. lets move it out

Reply:
I changed to: cd->endpoint[i]->ipc_config.type, like copier_free, move out always cause two same conditions.
Also, this part code is extracted from host_reset, need pay attention on the code sequence.